.circleci/config.yml

```diff
@@ -1,91 +1,72 @@
-        bundle exec hatchet ci:setup
+executors:
+  rust:
+    machine:
+      image: ubuntu-2004:202111-02
 
-jobs:
-  shellcheck:
-    docker:
-      - image: 'cimg/base:stable'
+commands:
+  restore-cargo-cache:
     steps:
-      - checkout
-      - shellcheck/install
-      - run:
-          name: Check build,compile,detect,release and bash_functions.sh with shellcheck
-          command: |
-            shellcheck -x bin/build bin/compile bin/detect bin/release bin/support/bash_functions.sh
-  unit:
-    docker:
-      - image: circleci/ruby:2.7
+      - restore_cache:
+          keys:
+            # CIRCLE_JOB is the job name, eg 'lint', ensuring caches are job-specific.
+            - cargo-{{ .Environment.CIRCLE_JOB }}-{{ checksum "Cargo.lock" }}-{{ checksum ".circleci/config.yml" }}
+  save-cargo-cache:
     steps:
-      - checkout
-      - ruby/install-deps
-      - <<: *hatchet_setup
-      - <<: *unit
-  hatchet:
-    docker:
-      - image: circleci/ruby:2.7
+      - save_cache:
+          key: cargo-{{ .Environment.CIRCLE_JOB }}-{{ checksum "Cargo.lock" }}-{{ checksum ".circleci/config.yml" }}
+          paths:
+            - ~/.cargo
+  install-rust:
     steps:
-      - checkout
-      - ruby/install-deps
-      - <<: *hatchet_setup
-      - <<: *hatchet
-  pack_cnb:
-    docker:
-      - image: circleci/ruby:2.7
+      - run:
+          name: "Install Rust"
+          command: |
+            curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs -o rustup.sh
+            chmod +x rustup.sh
+            ./rustup.sh -y
+
+jobs:
+  lint:
+    executor: rust
     steps:
       - checkout
-      - setup_remote_docker:
-          version: 19.03.13
-      - pack/install-pack:
-          version: 0.23.0
-      - ruby/install-deps
-      - <<: *hatchet_setup
-      - <<: *pack_cnb
-  docker_commands:
-    docker:
-      - image: circleci/ruby:2.7
+      - restore-cargo-cache
+      - install-rust
+      - run:
+          name: Clippy
+          # Using --all-targets so tests are checked and --deny to fail on warnings.
+          command: |
+            cargo clippy --all-targets --all-features --locked -- --deny warnings
+      - run:
+          name: rustfmt
+          command: cargo fmt -- --check --verbose
+      - save-cargo-cache
+
+  test:
+    executor: rust
     steps:
       - checkout
-      - setup_remote_docker:
-          version: 19.03.13
+      - restore-cargo-cache
+      - install-rust
       - pack/install-pack:
           version: 0.23.0
-      - ruby/install-deps
-      - <<: *hatchet_setup
-      - <<: *docker_commands
+      - run:
+          name: Install musl
+          command: sudo apt-get update && sudo apt-get install musl-tools --no-install-recommends
+      - run:
+          name: Add musl target
+          command: rustup target add x86_64-unknown-linux-musl
+      - run:
+          name: Run tests
+          command: RUST_BACKTRACE=1 cargo test --all-features --locked -- --include-ignored
```

Comment:
If you wanted to split this out into separate `unit-tests` and `integration-tests` jobs (where the former can skip the expensive musl/pack/... setup), then the unit tests variant can drop the `--include-ignored`, and the integration test variant instead switch to `-- --ignored`.

See:
https://github.com/heroku/procfile-cnb#test